resource/js/vocab-search.js

```diff
@@ -6,49 +6,246 @@ const vocabSearch = Vue.createApp({
+          result.pageUrl = window.SKOSMOS.vocab + '/' + window.SKOSMOS.lang + '/page?'
+          const urlParams = new URLSearchParams({ uri: result.uri })
+          result.pageUrl += urlParams.toString()
+        }
+        // render search result renderedTypes
+        if (result.type.length > 1) { // remove the type for SKOS concepts if the result has more than one type
+          result.type.splice(result.type.indexOf('skos:Concept'), 1)
+        }
+        // use the translateType function to map translations for the type IRIs
+        result.renderedType = result.type.map(this.translateType).join(', ')
+      })
+
+      if (this.renderedResultsList.length === 0) { // show no results message
+        this.renderedResultsList.push({
+          prefLabel: this.msgs['No results'],
+          lang: window.SKOSMOS.lang
+        })
+      }
+      const element = document.getElementById('search-autocomplete-results')
+      element.classList.add('show')
+    },
+    hideDropdown () {
+      const element = document.getElementById('search-autocomplete-results')
+      element.classList.remove('show')
     },
     gotoSearchPage () {
       if (!this.searchTerm) return
 
       const currentVocab = window.SKOSMOS.vocab + '/' + window.SKOSMOS.lang + '/'
       const vocabHref = window.location.href.substring(0, window.location.href.lastIndexOf(window.SKOSMOS.vocab)) + currentVocab
-      let langParam = '&clang=' + window.SKOSMOS.content_lang
-      if (this.selectedLanguage === 'all') langParam += '&anylang=on'
-      const searchUrl = vocabHref + 'search?q=' + this.searchTerm + langParam
+      const searchUrlParams = new URLSearchParams({ clang: window.SKOSMOS.content_lang, q: this.searchTerm })
+      if (this.selectedLanguage === 'all') searchUrlParams.set('anylang', 'on')
+      const searchUrl = vocabHref + 'search?' + searchUrlParams.toString()
       window.location.href = searchUrl
     },
     changeLang () {
       window.SKOSMOS.content_lang = this.selectedLanguage
-      // TODO: Impelemnt partial page load to change content according to the new content language
+      // TODO: Implement (a normal) page load to change content according to the new content language
+    },
+    resetSearchTermAndHideDropdown () {
+      this.searchTerm = ''
+      this.renderedResultsList = []
+      this.hideDropdown()
+    },
+    /*
+     * Hide the autocomplete result list if the user clicks outside the list
+     */
+    onClickOutside (event) {
+      const listener = document.getElementById('headerbar-search')
+      // Check if the clicked element is outside your element
+      if (listener && !listener.contains(event.target)) {
+        this.hideDropdown()
+      }
+    },
+    /*
+     * Show the existing autocomplete list if it was hidden by onClickOutside()
+     */
+    showAutoComplete () {
+      const element = document.getElementById('search-autocomplete-results')
+      element.classList.add('show')
```

Comment:
Instead of adding class directly, I would use a variable to control dropdown visibility, for example:
```suggestion
      this.showDropdown = true
```